### PR TITLE
fix: informative Stubs when document is invalid

### PR DESCRIPTION
**Pull request by catosaurusrex2003**
**Description**
Diagnostics with severity 0 are displayed as errors.

Before 
(Invalid document 👇 with syntax error)
image

After
(Invalid document 👇 with syntax error)
image

(document with spec specific errors 👇)
image

**Related Issue**
 #1048 
#874

### CODE REVIEW

File: library/src/containers/Error/Error.tsx [hunk 1]
```diff
@@ -10,12 +10,13 @@ const renderErrors = (errors: ValidationError[]): React.ReactNode => {
...
-      if (!singleError?.title || !singleError.location) {
+      if (!singleError?.title) {
         return null;
       }
       return (
-        <div key={index} className="flex">
-          <span>{`${singleError.location.startLine}.`}</span>
+        <div key={index} className="flex gap-2">
+          {/* <span>{`${singleError?.location?.startLine}.${singleError?.location?.startColumn}`}</span> */}
+          {/* <span> </span> */}
```
Comment (DhairyaMajmudar) [inline library/src/containers/Error/Error.tsx:19]: Please remove these comments

**Comment by catosaurusrex2003** [comment]
The line in diagnostics being returned by the parser is not accurate when there are `multi-line strings` or when the lines move back indentations.

@derberg should we include the line number in the errors ?

example: below error should be on line 12, but it is showing on line 7 cuz it ignored the multi-lined string of `/info/description`
image

**Comment by catosaurusrex2003** [comment]
I upgraded the package `@asyncapi/parser` from `3.1.0` -> `3.3.0`
and now the error lines are reliable with just an offset of 1

**Review by AceTheCreator — APPROVED** [review]
LGTM 🚀

**Comment by AceTheCreator** [comment]
@catosaurusrex2003, i noticed something after merging this PR that the error line seem not to be acccurate for some of the errors just as seen below. It should be line 3 not line 4 🤔
AsyncAPI-React-Playground-App-10-07-2024_06_34_PM

**Comment by AceTheCreator** [comment]
Wanna fix this in a follow-up PR?

**Comment by catosaurusrex2003** [comment]
@AceTheCreator I would love to work on this. But I cant seem to reproduce this issue again.

i tried the below on [https://asyncapi.github.io/asyncapi-react/](https://asyncapi.github.io/asyncapi-react/)
image
and it is giving correct line number error

could it be you are running it locally without reinstalling the `nodemodules` again, cuz our `@asyncapi/parser` dependency version changed in the pr.

**Comment by AceTheCreator** [comment]
Yea, that was the reason 😄. Everything looks good 🚀

Thanks for the fix 👍🏾

**Pull request merged**